oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,343 @@
+
+   * - OEP
+     - Link to the doc in the following format::
+
+        :doc:`OEP-67 <oep-0067-bp-tools-and-technology>`
+
+   * - Title
+     - Tools and Technology Standards
+   * - Last Modified
+     - 2023-09-05
+   * - Authors
+     - Feanil Patel <feanil@axim.org>
+   * - Arbiter
+     - Braden MacDonald  <braden@opencraft.com>
+   * - Status
+     - Accepted
+   * - Type
+     - Best Practice
+   * - Created
+     - 2023-09-05
+   * - Review Period
+     - 2023-09-05 - 2023-10-10
+   * - Resolution
+     - * https://github.com/openedx/open-edx-proposals/pull/518
+   * - References
+     - :doc:`oep-0011-bp-FED-technology`
+
+Abstract
+********
+
+The Open edX Platform is built using many 1st and 3rd party tools and technologies.
+This document covers the current recommendations for which tools and technologies should be used.
+
+Motivation
+**********
+
+The Open edX platform has a lot of 1st and 3rd party tools and technology that it uses.
+The problems we're trying to solve are:
+
+    * It is difficult to know what the current best practice for a given area.
+
+    * It is unclear how to go about the discussion and decision making process
+      to change a best practice.
+
+:doc:`OEP-11 <oep-0011-bp-FED-technology>` serves as a great example for the
+usefulness of this kind of OEP but limits itself to be relevant only to
+frontend technologies.  This OEP would supersede :doc:`OEP-11
+<oep-0011-bp-FED-technology>` and encompass not only frontend technolog but any
+technology that:
+
+    * Is considered a best practice for the Open edX Platform
+
+    * Is relevant to a large subset of the Open edX Platform
+
+Specification
+*************
+
+.. note:: We'll fill this out before the OEP lands but leaving mostly blank for initial feedback.
+
+General Technology Selection
+============================
+
+This technology is not specific to frontend or backend code.
+
+#. **Codecov should be used to report code coverage by unit tests**
+
+   **Rationale**: It is important to measure the amount of code covered by our
+   automated test suites. By striving for a high level of test coverage, we can
+   reduce the number of bugs that can only be found via manual testing, 
+   and by using codecov to run coverage in CI,
+   contributors are automatically reminded to include tests for any new code.
+
+   The Open edX community has standardized around `Codecov`_ as a coverage reporting service.
+
+#. **Use Github actions for CI Testing**
+
+   **Rationale**: As a healthy open source project we need good continouse integration
+   testing to prevent the introduction of obvious bugs into the platform code.  The Github Actions
+   tool should be used to perform Continuous Integration(CI) testing for all Open edX repositories
+   that perform any sort of software testing.
+
+Frontend Technology Selection
+=============================
+
+.. note:: 
+
+   The phrase “frontend” is used to mean any part of the platform that is shown to
```

Comment:
Do we actually standardise on "frontend", one word, no punctuation? It's fine if we do, that's just sufficiently unusual that it will probably trigger various spellcheckers and auto-correct tools.

Reply:
As far as I know we do standardize on "frontend." It does seem to trigger spellcheckers, but it also seems to be a very common spelling. For example, [the wiki page for "Frontend and backend"](https://en.wikipedia.org/wiki/Frontend_and_backend) notes:
> the terms frontend and backend (sometimes written as back end or back-end)